store/sqlstore/link_metadata_store.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type SqlLinkMetadataStore struct {
+	SqlStore
+}
+
+func NewSqlLinkMetadataStore(sqlStore SqlStore) store.LinkMetadataStore {
+	s := &SqlLinkMetadataStore{sqlStore}
+
+	for _, db := range sqlStore.GetAllConns() {
+		table := db.AddTableWithName(model.LinkMetadata{}, "LinkMetadata").SetKeys(false, "Hash")
+		table.ColMap("URL").SetMaxSize(512)
```

Comment:
Where does 512 come from? Is that enough for URL? I thought URLs could go up to about 2000 characters.

Reply:
Good point. The max I've found is 2083, supposedly, and since I'm not trying to make the URL the primary key any more, it probably makes sense to match that. I'll change that.

Reply:
I've changed it to 2048 to keep it a round number. In practice, it may go longer, but there shouldn't be any negative side effects of longer URLs outside of maybe not caching data for a really long link.